docs/source/tutorial_optimization.rst

```diff
@@ -0,0 +1,894 @@
+    reference_plot_mean_z_lim = 0.5*(reference_plot.get_zlim()[0] + reference_plot.get_zlim()[1])
+    
+    for mu in reference_minimization_data['evaluation_points']:
+        addplot_xy_point_as_bar(reference_plot, mu[0], mu[1], color='green',
+                                z_range=(reference_plot.get_zlim()[0], reference_plot_mean_z_lim))
+    
+    for mu in RB_minimization_data['evaluation_points']:
+        addplot_xy_point_as_bar(reference_plot, mu[0], mu[1], color='orange',
+                               z_range=(reference_plot_mean_z_lim, reference_plot.get_zlim()[1]))
+
+
+
+Computing the gradient of the objective functional
+--------------------------------------------------
+
+A major issue of using finite differences for computing the gradient of
+the objective functional is the number of evaluations of the objective
+functional. In the FOM example from above we saw that
+many evaluations of the model were only due to the
+computation of the finite differences. If the problem is more complex
+and the mesh is finer, this can lead to a serious waste of
+computational time. Also from an optimizational point of view it is
+always better to compute the true gradient of the objective functional.
+
+For computing the gradient of the linear objective functional
+:math:`\mathcal{J}(\mu)`, we can write for every direction
+:math:`i= 1, \dots, P`
+
+.. math::
+
+
+   \begin{align} \label{gradient:sens} \tag{1}
+   d_{\mu_i} \mathcal{J}(\mu) = \partial_{\mu_i} J(u_{\mu}, \mu) + \partial_u J(u_{\mu}, \mu)[d_{\mu_i} u_{\mu}] 
+      =   \partial_{\mu_i} J(u_{\mu}, \mu) + J(d_{\mu_i} u_{\mu}, \mu)
+   \end{align}
+
+Thus, we need to compute the derivative of the
+solution :math:`u_{\mu}` (also called sensitivity). For this, we need to
+solve another equation: Find :math:`d_{\mu_i} u_{\mu} \in V`, such that
+
+.. math::
+
+    \label{sens} \tag{2}
+   a_\mu(d_{\mu_i} u_{\mu}, v) = \partial_{\mu_i} r_\mu^{\text{pr}}(u_{\mu})[v] \qquad \qquad \forall v \in V
+
+where :math:`r_\mu^{\text{pr}}` denotes the residual of the primal
+equation, i.e.
+
+.. math::
+        r_\mu^{\text{pr}(u)[v] := l_\mu(v) - a_\mu(u, v) &&\text{for all }v \in V
+
+A major issue of this approach is that the computation of the
+full gradient requires :math:`P` solutions of :math:`\eqref{sens}`.
+Especially for high dimensional parameter spaces, we can instead use an
+adjoint approach to reduce the computational cost to only one solution
+of an additional problem.
+
+The adjoint approach relies on the Lagrangian of the objective
+functional
+
+.. math::
+
+
+   \mathcal{L}(u, \mu, p) = J(u, \mu) + r_\mu^{\text{pr}}(u, p)
+
+where :math:`p \in V` is the adjoint variable. Deriving optimality
+conditions for :math:`\mathcal{L}`, we end up with the dual equation:
+Find :math:`p_{\mu} \in V`, such that
+
+.. math::
+
+    \label{dual} \tag{3}
+   a_\mu(v, p_\mu) = \partial_u J(u_\mu, \mu)[v]
+   = J(v, \mu)
+
+Note that in our case, we then have
```

Comment:
Perhaps mention quickly why it vanishes here (spaces and so on).

Reply:
I wrote half a sentence more